spec/operators/expand-spec.js

```diff
@@ -27,14 +30,214 @@ describe('Observable.prototype.expand()', function () {
+    var values = {
+      a: 1,
+      b: 1 + 1, // a + a,
+      c: 2 + 2, // b + b,
+      d: 4 + 4, // c + c,
+      e: 8 + 8, // d + d
+      u: 10,
+      v: 20,  // u + u
+      x: 40,  // v + v
+      y: 80,  // x + x
+      z: 160, // y + y
+    };
+    var e1 =   hot('a-u|', values);
+    var e2shape =  '---(z|)';
+    //                 ---(z|)
+    //                    ---(z|)
+    //                       ---(z|)
+    //                          ---(z|)
+    //                             ---(z|)
+    //                                ---(z|)
+    //                                   ---(z|)
+    // Notice how for each column, there is at most 1 `-` character.
+    var e1subs =   '^                       !    ';
+    var expected = 'a--u--b--v--c--x--d--y--(ez|)';
+    var concurrencyLimit = 1;
+
+    var result = e1.expand(function (x) {
+      if (x === 16 || x === 160) {
+        return Observable.empty();
+      }
+      return cold(e2shape, { z: x + x });
+    }, concurrencyLimit);
+
+    expectObservable(result).toBe(expected, values);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+  });
+
+  it('should allow configuring the concurrency limit parameter to 2', function () {
+    var values = {
+      a: 1,
+      b: 1 + 1, // a + a,
+      c: 2 + 2, // b + b,
+      u: 10,
+      v: 20,  // u + u
+      x: 40,  // v + v
+    };
+    var e1 =   hot('a---au|', values);
+    var e2shape =  '------(z|)';
+    //                  ------(z|)
+    //                    ------(z|)
+    //                        ------(z|)
+    //                          ------(z|)
+    //                              ------(z|)
+    //                                ------(z|)
+    // Notice how for each column, there is at most 2 `-` characters.
+    var e1subs =   '^                     !   ';
+    var expected = 'a---a-u---b-b---v-(cc)(x|)';
+    var concurrencyLimit = 2;
+
+    var result = e1.expand(function (x) {
+      if (x === 4 || x === 40) {
+        return Observable.empty();
+      }
+      return cold(e2shape, { z: x + x });
+    }, concurrencyLimit);
+
+    expectObservable(result).toBe(expected, values);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+  });
+
+  it('should ignore concurrency limit if it is not passed', function () {
+    var values = {
+      a: 1,
+      b: 1 + 1, // a + a,
+      c: 2 + 2, // b + b,
+      d: 4 + 4, // c + c,
+      e: 8 + 8, // d + d
+      u: 10,
+      v: 20,  // u + u
+      x: 40,  // v + v
+      y: 80,  // x + x
+      z: 160, // y + y
+    };
+    var e1 =   hot('a-u|              ', values);
+    var e1subs =   '^             !   ';
+    var e2shape =  '---(z|)           ';
+    var expected = 'a-ub-vc-xd-ye-(z|)';
+    var concurrencyLimit = 100;
+
+    var result = e1.expand(function (x) {
+      if (x === 16 || x === 160) {
+        return Observable.empty();
+      }
+      return cold(e2shape, { z: x + x });
+    }, concurrencyLimit);
```

Comment:
I meant that if the concurrency limit is high enough, then the behavior should be the same as not specifying it at all (which means Infinity). It's important to test because there is an `if` statement that only runs when the parameter is finite. We want to be sure "very big number" works the same as "infinite".